mobile-app/app/screens/AppNavigator/screens/Portfolio/components/CreateOrEditAddressLabelFormV2.tsx

```diff
@@ -0,0 +1,298 @@
+import { memo, useCallback, useEffect, useState } from 'react'
+import { StackScreenProps } from '@react-navigation/stack'
+import { BottomSheetWithNavRouteParam } from '@components/BottomSheetWithNav'
+import { ThemedScrollViewV2, ThemedText, ThemedTextV2 } from '@components/themed'
+import { BottomSheetScrollView } from '@gorhom/bottom-sheet'
+import { Platform, View } from 'react-native'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { RandomAvatar } from './RandomAvatar'
+import { WalletTextInput } from '@components/WalletTextInput'
+import { LabeledAddress, LocalAddress } from '@store/userPreferences'
+import { fromAddress } from '@defichain/jellyfish-address'
+import { useNetworkContext } from '@shared-contexts/NetworkContext'
+import { useSelector } from 'react-redux'
+import { RootState } from '@store'
+import { authentication, Authentication } from '@store/authentication'
+import { MnemonicStorage } from '@api/wallet/mnemonic_storage'
+import { useWalletNodeContext } from '@shared-contexts/WalletNodeProvider'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { useAppDispatch } from '@hooks/useAppDispatch'
+import { WalletTextInputV2 } from '@components/WalletTextInputV2'
+import { SubmitButtonGroupV2 } from '@components/SubmitButtonGroupV2'
+import { useThemeContext } from '@shared-contexts/ThemeProvider'
+
+export interface CreateOrEditAddressLabelFormProps {
+  title: string
+  isAddressBook: boolean
+  address?: string
+  addressLabel?: LocalAddress
+  onSaveButtonPress: (labelAddress: LabeledAddress) => void
+}
+
+type Props = StackScreenProps<BottomSheetWithNavRouteParam, 'CreateOrEditAddressLabelFormProps'>
+
+export const CreateOrEditAddressLabelFormV2 = memo(({
+  route,
+  navigation
+}: Props): JSX.Element => {
+  // const { isLight } = useThemeContext()
+  const {
+    title,
+    isAddressBook,
+    address,
+    addressLabel,
+    onSaveButtonPress
+  } = route.params
+  const { isLight } = useThemeContext()
+  const [labelInput, setLabelInput] = useState(addressLabel?.label)
+  const [addressInput, setAddressInput] = useState<string | undefined>()
+  const bottomSheetComponents = {
+    mobile: BottomSheetScrollView,
+    web: ThemedScrollViewV2
+  }
+  const ScrollView = Platform.OS === 'web' ? bottomSheetComponents.web : bottomSheetComponents.mobile
+  const { networkName } = useNetworkContext()
+  const addressBook = useSelector((state: RootState) => state.userPreferences.addressBook)
+  const [labelInputErrorMessage, setLabelInputErrorMessage] = useState('')
+  const [addressInputErrorMessage, setAddressInputErrorMessage] = useState('')
+  const [labelInputLength, setLabelInputLength] = useState(0)
+
+  useEffect(() => {
+    if (labelInput !== undefined) {
+      setLabelInputLength(labelInput.trim().length)
```

Comment:
Should it count spaces in between? e.g `t e s t` - this should be 4 chars or 8?

Reply:
should be 7 chars, since trim only applies to spacing of start and end